File: typescript/armstrong-numbers/armstrong-numbers.ts
export function isArmstrongNumber(number: number | bigint): boolean {
  let str = number.toString();
  let sum: bigint = BigInt(0);
  for (let i = 0; i < str.length; i++) {
    let digit = BigInt(str.charAt(i));
    sum += digit ** BigInt(str.length);
  }
  return sum === BigInt(number);
}

File: solutions/dart/armstrong-numbers/2/lib/armstrong_numbers.dart
class ArmstrongNumbers {
  bool isArmstrongNumber(String str) {
    BigInt res = BigInt.zero;
    int length = str.length;

    for (int i = 0; i < length; i++) {
      BigInt digit = BigInt.parse(str[i]);
      res += _bigIntPow(digit, length);
    }

    return res == BigInt.parse(str);
  }

  BigInt _bigIntPow(BigInt base, int exponent) {
    BigInt result = BigInt.one;
    for (int i = 0; i < exponent; i++) {
      result *= base;
    }
    return result;
  }
}
